feat(app-shell): add ability to `skipTokenRetry` with variable (#1755)
commercetools/merchant-center-application-kit: .changeset/chilled-snakes-thank.md

---
"@commercetools-frontend/application-shell": minor	
---

> feat(app-shell): add ability to `skipTokenRetry` via context.

Reviewer: ```suggestion
feat(app-shell): add ability to `skipTokenRetry` for Apollo queries.
```

> The commercetools Platform token is renewed once whenever it expired. This is an automatic mechanism which retries a failed request responding with `401` once based on the GraphQL target.

Reviewer: ```suggestion
The Merchant Center API Gateway assigns a commercetools platform API token in order to access the commercetools HTTP APIs. The access token eventually expires, causing requests to fail with HTTP `401`.
A Custom Application comes with a built-in mechanism to automatically retries unauthorized requests by forcing the Merchant Center API Gateway to assign a new valid API token. This retry mechanism is configured for Apollo queries for certain GraphQL APIs.
```

> It is however useful for some requests to disable this mechanism to avoid uncalled-for network requests. We introduced a `skipTokenRetry` property on the GraphQL `context` for this.

Reviewer: ```suggestion
However, it is useful for some requests to disable this mechanism to avoid uncalled-for network requests. This is done by specifying a `skipTokenRetry` property on the Apollo query `context` object.
```
